### PR TITLE
#2799893: Port to D8

### PR DESCRIPTION
**Pull request by Stolexiy**

### CODE REVIEW

File: rpt.info.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name: Registration Password Token
+type: module
+description: 'Provides user password token during registration.'
+package: 'User Access & Authentication'
+version: VERSION
```
Comment (andypost) [inline rpt.info.yml]: please remove

File: rpt.module [hunk 2]
```diff
@@ -1,51 +1,72 @@
 <?php
```
Comment (andypost) [inline rpt.module]: needs standard doc block

```
* @file
* ....
```

File: rpt.module [hunk 3]
```diff
@@ -62,7 +83,7 @@ function rpt_token_info() {
 /**
  * Implements hook_tokens().
  */
-function rpt_tokens($type, $tokens, array $data = array(), array $options = array()) {
+function rpt_tokens($type, $tokens, array $data = array(), array $options = array(), \Drupal\Core\Render\BubbleableMetadata $bubbleable_metadata) {
```
Comment (andypost) [inline rpt.module]: please use `use` to not display full namespace in code

File: rpt.module [hunk 4]
```diff
@@ -1,51 +1,72 @@
...
+/**
+ * Submit handler for 'user_form' form.
+ *
+ * @see: rpt_form_user_form_alter()
  */
-function rpt_user_profile_form_submit(&$form, &$form_state) {
-  $form_state['user']->password = $form_state['values']['pass'];
+function rpt_user_form_submit($form, \Drupal\Core\Form\FormStateInterface $form_state) {
+  $pass = $form_state->getValue('pass');
+  // @todo: Assign pass to the user object.
```
Comment (andypost) [inline rpt.module]: use http://cgit.drupalcode.org/drupal/tree/core/modules/contact/contact.module#n176

File: rpt.module (hunk 2 above)
Comment (Shdulin) [inline rpt.module]: Added

File: rpt.install [hunk 5]
```diff
@@ -9,6 +9,5 @@
  * Implements hook_uninstall().
  */
 function rpt_uninstall() {
-  variable_del('rpt_password_generate');
-  variable_del('rpt_password_length');
+  \Drupal::service('config.factory')->getEditable('rpt.settings')->delete();
```
Comment (andypost) [inline rpt.install]: this will be done automatically by CMI, just remove that file

File: README.txt [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-1. Extract module into /sites/all/modules folder.
+1. Extract module into /modules folder.
 2. Enable module on /admin/modules page.
 
 
 USAGE
 ==========
 
-Go to /admin/config/people/accounts/settings page and use new token [user:password] in mail texts.
+Go to /admin/config/people/accounts page and use new token [user:password] in mail texts.
```
Comment (andypost) [inline README.txt:11]: URL really changed?

Comment (Stolexiy) [reply README.txt:11]: Yes. It was changes, as I see http://storage8.static.itmages.com/i/16/0919/h_1474274321_2525140_de22ef0f23.png

File: rpt.install (hunk 5 above)
Comment (Stolexiy) [reply rpt.install]: Removed

File: config/install/rpt.settings.yml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+password_generate: false
+password_length: 10
```
Comment (andypost) [inline config/install/rpt.settings.yml:2]: last issue is that you need config schema for that

Comment (andypost) [reply config/install/rpt.settings.yml:2]: Suppose following `config/schema/rpt.schema.yml`

```
rpt.settings:
  type: config_object
  label: 'RPT settings'
  mapping:
    password_generate:
      type: boolean
      label: 'Generate password automatically'
    password_length:
      type: integer
      label: 'Generated password length'
```

PS: http://hojtsy.hu/files/ConfigSchemaCheatSheet1.5.pdf

Comment (Stolexiy) [reply config/install/rpt.settings.yml:2]: Added schema

**Review by andypost — APPROVED** [review]

**Pull request merged**